### ISSUE TITLE
Error in Deployment of TF Serving Batch Transform

### ISSUE THREAD

**Issue by Ankur3107** [aws/sagemaker-tensorflow-serving-container#93]
Reference: MLFW-2729

Hi AWS,

I have a Tensorflow model in S3 and want to make TF Serving Batch transform job. 

ECS images used: *520713654638.dkr.ecr.ap-south-1.amazonaws.com/sagemaker-tensorflow-serving:1.13-gpu*

**Error:**

- data-log:  
`<head><title>502 Bad Gateway</title></head>#015`

- docker-log:
```
INFO:__main__:starting services
INFO:__main__:using default model name: Servo
INFO:__main__:tensorflow serving model config: 
model_config_list: {
  config: {
    name: "Servo",
    base_path: "/opt/ml/model/export/Servo",
    model_platform: "tensorflow"
  }
}

INFO:__main__:nginx config: 
load_module modules/ngx_http_js_module.so;
worker_processes auto;
daemon off;
pid /tmp/nginx.pid;
error_log  /dev/stderr error;
worker_rlimit_nofile 4096;
events {
  worker_connections 2048;
}
http {
  include /etc/nginx/mime.types;
  default_type application/json;
  access_log /dev/stdout combined;
  js_include tensorflow-serving.js;

  upstream tfs_upstream {
    server localhost:10001;
  }

  upstream gunicorn_upstream {
    server unix:/tmp/gunicorn.sock fail_timeout=1;
  }

  server {
    listen 8080 deferred;
    client_max_body_size 0;
    client_body_buffer_size 100m;
    subrequest_output_buffer_size 100m;

    set $tfs_version 1.13;
    set $default_tfs_model Servo;

    location /tfs {
        rewrite ^/tfs/(.*) /$1  break;
        proxy_redirect off;
        proxy_pass_request_headers off;
        proxy_set_header Content-Type 'application/json';
        proxy_set_header Accept 'application/json';
        proxy_pass http://tfs_upstream;
    }

    location /ping {
        js_content ping;
    }

    location /invocations {
        js_content invocations;
    }

    location ~ ^/models/(.*)/invoke {
        js_content invocations;
    }

    location /models {
        proxy_pass http://gunicorn_upstream/models;
    }

    location / {
        return 404 '{"error": "Not Found"}';
    }

    keepalive_timeout 3;
  }
}

INFO:__main__:tensorflow version info:
TensorFlow ModelServer: 1.13.0-rc1+dev.sha.f16e777
TensorFlow Library: 1.13.1
INFO:__main__:tensorflow serving command: tensorflow_model_server --port=10000 --rest_api_port=10001 --model_config_file=/sagemaker/model-config.cfg 
INFO:__main__:started tensorflow serving (pid: 10)
INFO:__main__:nginx version info:
nginx version: nginx/1.16.1
built by gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.11) 
built with OpenSSL 1.0.2g  1 Mar 2016
TLS SNI support enabled
configure arguments: --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp --http-scgi-temp-path=/var/cache/nginx/scgi_temp --user=nginx --group=nginx --with-compat --with-file-aio --with-threads --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-mail --with-mail_ssl_module --with-stream --with-stream_realip_module --with-stream_ssl_module --with-stream_ssl_preread_module --with-cc-opt='-g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fPIC' --with-ld-opt='-Wl,-Bsymbolic-functions -fPIE -pie -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie'
INFO:__main__:started nginx (pid: 12)
2019-11-21 08:34:19.173438: I tensorflow_serving/model_servers/server_core.cc:461] Adding/updating models.
2019-11-21 08:34:19.173472: I tensorflow_serving/model_servers/server_core.cc:558]  (Re-)adding model: Servo
169.254.255.130 - - [21/Nov/2019:08:34:19 +0000] "GET /ping HTTP/1.1" 200 3 "-" "Go-http-client/1.1"
169.254.255.130 - - [21/Nov/2019:08:34:19 +0000] "GET /execution-parameters HTTP/1.1" 404 22 "-" "Go-http-client/1.1"
2019/11/21 08:34:19 [error] 15#15: *3 connect() failed (111: Connection refused) while connecting to upstream, client: 169.254.255.130, server: , request: "POST /invocations HTTP/1.1", subrequest: "/v1/models/Servo:predict", upstream: "http://127.0.0.1:10001/v1/models/Servo:predict", host: "169.254.255.131:8080"
2019/11/21 08:34:19 [error] 15#15: *3 connect() failed (111: Connection refused) while connecting to upstream, client: 169.254.255.130, server: , request: "POST /invocations HTTP/1.1", subrequest: "/v1/models/Servo:predict", upstream: "http://127.0.0.1:10001/v1/models/Servo:predict", host: "169.254.255.131:8080"
169.254.255.130 - - [21/Nov/2019:08:34:19 +0000] "POST /invocations HTTP/1.1" 502 157 "-" "Go-http-client/1.1"
2019-11-21 08:34:19.273915: I tensorflow_serving/core/basic_manager.cc:739] Successfully reserved resources to load servable {name: Servo version: 1}
2019-11-21 08:34:19.273965: I tensorflow_serving/core/loader_harness.cc:66] Approving load for servable version {name: Servo version: 1}
2019-11-21 08:34:19.273985: I tensorflow_serving/core/loader_harness.cc:74] Loading servable version {name: Servo version: 1}
2019-11-21 08:34:19.274012: I external/org_tensorflow/tensorflow/contrib/session_bundle/bundle_shim.cc:363] Attempting to load native SavedModelBundle in bundle-shim from: /opt/ml/model/export/Servo/1
2019-11-21 08:34:19.274034: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:31] Reading SavedModel from: /opt/ml/model/export/Servo/1
2019-11-21 08:34:19.381053: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:54] Reading meta graph with tags { serve }
2019-11-21 08:34:19.512732: I external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-11-21 08:34:19.766363: I external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-11-21 08:34:19.767135: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8755
pciBusID: 0000:00:1e.0
totalMemory: 11.17GiB freeMemory: 11.11GiB
2019-11-21 08:34:19.767169: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-11-21 08:34:20.178943: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-11-21 08:34:20.178986: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2019-11-21 08:34:20.178996: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2019-11-21 08:34:20.179129: I external/org_tensorflow/tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10751 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0, compute capability: 3.7)
2019-11-21 08:34:20.381256: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:182] Restoring SavedModel bundle.
2019-11-21 08:34:21.653023: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:132] Running initialization op on SavedModel bundle.
2019-11-21 08:34:23.134589: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:285] SavedModel load for tags { serve }; Status: success. Took 3860532 microseconds.
2019-11-21 08:34:23.134672: I tensorflow_serving/servables/tensorflow/saved_model_warmup.cc:101] No warmup data file found at /opt/ml/model/export/Servo/1/assets.extra/tf_serving_warmup_requests
2019-11-21 08:34:23.134790: I tensorflow_serving/core/loader_harness.cc:86] Successfully loaded servable version {name: Servo version: 1}
2019-11-21 08:34:23.137078: I tensorflow_serving/model_servers/server.cc:313] Running gRPC ModelServer at 0.0.0.0:10000 ...
[warn] getaddrinfo: address family for nodename not supported
2019-11-21 08:34:23.138567: I tensorflow_serving/model_servers/server.cc:333] Exporting HTTP/REST API at:localhost:10001 ...
[evhttp_server.cc : 237] RAW: Entering the event loop ...
```

#### My Observation:

During server(docker run) loading tf model, /ping request came and Nginx gave 200 response. Then /invocations API called and it returned 502 because model was still in loading phase. (you can see after some sec model get loaded i.e [evhttp_server.cc: 237] RAW: Entering the event loop ...) 

- Is this because of docker/Nginx? (I used aws tf serving container)
 

#### Can you help me with this.

Regards,
Ankur

**Comment by ChoiByungWook** [comment]
Hello @Ankur3107,

Apologies for the late response.

Thank you for bringing this to our attention. 

Do you run into the same issue when you deploy to a regular endpoint instead of using batch?
Can you please provide your SageMaker code used to start the batch transform job?

Thanks!

**Comment by Ankur3107** [comment]
1. _Do you run into the same issue when you deploy to a regular endpoint instead of using batch?_

      `No, It is working fine with the endpoint.`

2. _Can you please provide your SageMaker code used to start the batch transform job?_

```
# Job configuration

- Instance type: ml.p2.xlarge
- Instance count: 1
- Max concurrent transforms: 1
- Max payload size (MB): 5
- Batch strategy: SingleRecord

# Input data configuration

- Split type: Line
- Content type: application/jsonlines

# Output data configuration

- Assemble with: Line

```

**Other parameters are default values.**

**Comment by ChoiByungWook** [comment]
@Ankur3107,

Awesome, thank you so much for confirming this.

This error you are seeing is related to an issue on our end. Most likely the issue is due to this configuration, which will return a 200 instead of waiting on TFS to finish loading the model: https://github.com/aws/sagemaker-tensorflow-serving-container/pull/82/files#diff-d144722fedacc45fd5f91c6b0313f415L16

I'll speak with the team on how to handle this and have also added created an internal ticket on our end to track this.

I apologize for the experience.

**Comment by RZachLamberty** [comment]
I have been experiencing this same behavior with the 1.15 framework container on a GPU instance. I also see broadly inconsistent behavior for multi-file batch transforms, with some runs completing successfully and others (most) failing for at least a handful of the input files for reasons outlined above (unresponsive boxes during model build)

**Comment by Ankur3107** [comment]
@RZachLamberty yes, you are right, I am also facing inconsistent behavior for multi-file batch transforms.
@ChoiByungWook can you update us regarding the same.

**Comment by laurenyu** [comment]
The fix has been merged and released for TFS 1.13. It will be released for TFS 1.15 in the near future.

**Comment by RZachLamberty** [comment]
great news @laurenyu . is there anywhere I can see the release state for 1.15 (or 1.14, assuming you'll be releasing that as well) so i can know when I am able to update?

**Comment by laurenyu** [comment]
Unfortunately, I can't give any more detail at the moment, but we'll update this thread when the fix has been released for later TFS versions.

**Comment by RZachLamberty** [comment]
@laurenyu sorry to be a pest but we are looking to deploy a model using sagemaker in the next two weeks and will have to make a different decision if 1.15 is not supported. do you have an order of magnitude estimate for how long this will be (days, weeks, months)? right now we are attempting to process 200 items and 150 fail before scoring

**Comment by awcchungster** [comment]
Hi @RZachLamberty , I'm the product manager for the Deep Learning Containers.

We just rereleased the 1.15 TFS containers. The docker image IDs can be found here:

763104351884.dkr.ecr.us-east-1.amazonaws.com/tensorflow-inference:1.15.0-gpu
763104351884.dkr.ecr.us-east-1.amazonaws.com/tensorflow-inference:1.15.0-cpu

**Comment by RZachLamberty** [comment]
thanks so much @awcchungster, @laurenyu, and team! Just tried it out and it works as expected!

fwiw during the startup (when the model is being built and the `/ping` calls fail), the cloudwatch logs are spammed with

```
169.254.255.130 - - [18/Jan/2020:22:42:00 +0000] "GET /ping HTTP/1.1" 502 157 "-" "Go-http-client/1.1"
2020/01/18 22:42:00 [error] 64#64: *398 no live upstreams while connecting to upstream, client: 169.254.255.130, server: , request: "GET /ping HTTP/1.1", subrequest: "/v1/models/model:predict", upstream: "http://tfs_upstream/v1/models/model:predict", host: "169.254.255.131:8080"
169.254.255.130 - - [18/Jan/2020:22:42:00 +0000] "GET /ping HTTP/1.1" 502 157 "-" "Go-http-client/1.1"
2020/01/18 22:42:00 [error] 64#64: *398 js: failed ping<html>#015
<head><title>502 Bad Gateway</title></head>#015
<body>#015
<center><h1>502 Bad Gateway</h1></center>#015
<hr><center>nginx/1.16.1</center>#015
</body>#015
</html>#015
```

it's not a problem ultimately (all files get scored) but if it were possible to turn off these logs I think it would avoid confusion for others seeing this happen during startup

**Comment by tbiker** [comment]
I am having a similar problem. I have a TensorFlow 2.4.1 model in S3 as per the model.tar.gz formats. The model.tar.tz includes inference.py to handle preparing an image payload for the model using input_handler. A requirements.txt file states that the json5 package should be loaded. It appears that all of this starts up correctly. 

In my python code, I use:

import numpy as np
import os
import boto3
import sagemaker
from sagemaker import get_execution_role
from sagemaker.tensorflow import TensorFlowModel

sagemaker_session = sagemaker.Session()
try:
    role = get_execution_role()
except ValueError:
    iam = boto3.client('iam')
    role = iam.get_role(RoleName='Sagemaker')['Role']['Arn']

region = sagemaker_session.boto_region_name
bucket = sagemaker_session.default_bucket()
prefix = 'cvmodel'
print('Region: {}'.format(region))
print('S3 URI: s3://{}/{}'.format(bucket, prefix))
print('Role:   {}'.format(role))

model = TensorFlowModel(model_data='s3://cvmodel/model.tar.gz',
                            role=role,
                            entry_point='inference.py',
                            framework_version="2.4.1"
                            )
transformer = model.transformer(instance_count=1,
                          instance_type='ml.m4.xlarge',
                          max_concurrent_transforms=1,
                          max_payload=1,
                          output_path='s3://cvmodel/results')
transformer.transform('s3://cvmodel/images', content_type='application/x-image')

The inference.py input_handler looks like this:

import base64
import io
import json
import requests

def input_handler(data, context):
    """ Pre-process request input before it is sent to TensorFlow Serving REST API
    Args:
        data (obj): the request data stream
        context (Context): an object containing request and configuration details
    Returns:
        (dict): a JSON-serializable dict that contains request body and headers
    """

    print('input handler called')
    if context.request_content_type == 'application/x-image':
        payload = data.read()
        encoded_image = base64.b64encode(payload).decode('utf-8')
        instance = [{"b64": encoded_image}]
        print('json image produced')
        return json.dumps({"instances": instance})
    else:
        _return_error(415, 'Unsupported content type "{}"'.format(context.request_content_type or 'Unknown'))

This is the log I receive after called the transform function and waiting about 4 minutes:

What is the problem here?

..........................INFO:__main__:starting services
INFO:tfs_utils:using default model name: Servo
INFO:tfs_utils:tensorflow serving model config: 
model_config_list: {
  config: {
    name: "Servo",
    base_path: "/opt/ml/model/export/Servo",
    model_platform: "tensorflow"
  }
}


INFO:__main__:using default model name: Servo
INFO:__main__:tensorflow serving model config: 
model_config_list: {
  config: {
    name: "Servo",
    base_path: "/opt/ml/model/export/Servo",
    model_platform: "tensorflow"
  }
}


INFO:__main__:tensorflow version info:
2021-03-14 23:56:24.251936: W external/org_tensorflow/tensorflow/core/profiler/internal/smprofiler_timeline.cc:460] Initializing the SageMaker Profiler.
2021-03-14 23:56:24.252074: W external/org_tensorflow/tensorflow/core/profiler/internal/smprofiler_timeline.cc:105] SageMaker Profiler is not enabled. The timeline writer thread will not be started, future recorded events will be dropped.
TensorFlow ModelServer: 2.4.0-rc4+dev.sha.no_git
TensorFlow Library: 2.4.1
INFO:__main__:tensorflow serving command: tensorflow_model_server --port=10000 --rest_api_port=10001 --model_config_file=/sagemaker/model-config.cfg --max_num_load_retries=0 
INFO:__main__:started tensorflow serving (pid: 12)
INFO:__main__:nginx config: 
load_module modules/ngx_http_js_module.so;

worker_processes auto;
daemon off;
pid /tmp/nginx.pid;
error_log  /dev/stderr error;

worker_rlimit_nofile 4096;

events {
  worker_connections 2048;
}

http {
  include /etc/nginx/mime.types;
  default_type application/json;
  access_log /dev/stdout combined;
  js_include tensorflow-serving.js;

  upstream tfs_upstream {
    server localhost:10001;
  }

  upstream gunicorn_upstream {
    server unix:/tmp/gunicorn.sock fail_timeout=1;
  }

  server {
    listen 8080 deferred;
    client_max_body_size 0;
    client_body_buffer_size 100m;
    subrequest_output_buffer_size 100m;

    set $tfs_version 2.4;
    set $default_tfs_model Servo;

    location /tfs {
        rewrite ^/tfs/(.*) /$1  break;
        proxy_redirect off;
        proxy_pass_request_headers off;
        proxy_set_header Content-Type 'application/json';
        proxy_set_header Accept 'application/json';
        proxy_pass http://tfs_upstream;
    }

    location /ping {
        proxy_pass http://gunicorn_upstream/ping;
    }

    location /invocations {
        proxy_pass http://gunicorn_upstream/invocations;
    }

    location /models {
        proxy_pass http://gunicorn_upstream/models;
    }

    location / {
        return 404 '{"error": "Not Found"}';
    }

    keepalive_timeout 3;
  }
}


INFO:__main__:installing packages from requirements.txt...
2021-03-14 23:56:24.637383: W external/org_tensorflow/tensorflow/core/profiler/internal/smprofiler_timeline.cc:460] Initializing the SageMaker Profiler.
2021-03-14 23:56:24.637508: W external/org_tensorflow/tensorflow/core/profiler/internal/smprofiler_timeline.cc:105] SageMaker Profiler is not enabled. The timeline writer thread will not be started, future recorded events will be dropped.
2021-03-14 23:56:24.640239: I tensorflow_serving/model_servers/server_core.cc:464] Adding/updating models.
2021-03-14 23:56:24.640284: I tensorflow_serving/model_servers/server_core.cc:587]  (Re-)adding model: Servo
2021-03-14 23:56:24.740563: I tensorflow_serving/util/retrier.cc:46] Retrying of Reserving resources for servable: {name: Servo version: 1} exhausted max_num_retries: 0
2021-03-14 23:56:24.740603: I tensorflow_serving/core/basic_manager.cc:740] Successfully reserved resources to load servable {name: Servo version: 1}
2021-03-14 23:56:24.740626: I tensorflow_serving/core/loader_harness.cc:66] Approving load for servable version {name: Servo version: 1}
2021-03-14 23:56:24.740645: I tensorflow_serving/core/loader_harness.cc:74] Loading servable version {name: Servo version: 1}
2021-03-14 23:56:24.740723: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:32] Reading SavedModel from: /opt/ml/model/export/Servo/1
2021-03-14 23:56:24.888950: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:55] Reading meta graph with tags { serve }
2021-03-14 23:56:24.889008: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:93] Reading SavedModel debug info (if present) from: /opt/ml/model/export/Servo/1
2021-03-14 23:56:24.889602: I external/org_tensorflow/tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
Collecting json5
  Downloading json5-0.9.5-py2.py3-none-any.whl (17 kB)
2021-03-14 23:56:25.373645: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:206] Restoring SavedModel bundle.
Installing collected packages: json5
Successfully installed json5-0.9.5
2021-03-14 23:56:25.435171: I external/org_tensorflow/tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2300050000 Hz
INFO:__main__:gunicorn command: gunicorn -b unix:/tmp/gunicorn.sock -k gevent --chdir /sagemaker --pythonpath /opt/ml/model/code -e TFS_GRPC_PORT=10000 -e SAGEMAKER_MULTI_MODEL=False -e SAGEMAKER_SAFE_PORT_RANGE=10000-10999 python_service:app
INFO:__main__:gunicorn version info:
gunicorn (version 20.0.4)
INFO:__main__:started gunicorn (pid: 47)
[2021-03-14 23:56:26 +0000] [47] [INFO] Starting gunicorn 20.0.4
[2021-03-14 23:56:26 +0000] [47] [INFO] Listening at: unix:/tmp/gunicorn.sock (47)
INFO:__main__:gunicorn server is ready!
[2021-03-14 23:56:26 +0000] [47] [INFO] Using worker: gevent
[2021-03-14 23:56:26 +0000] [51] [INFO] Booting worker with pid: 51
INFO:__main__:nginx version info:
nginx version: nginx/1.18.0
built by gcc 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 
built with OpenSSL 1.1.1  11 Sep 2018
TLS SNI support enabled
configure arguments: --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp --http-scgi-temp-path=/var/cache/nginx/scgi_temp --user=nginx --group=nginx --with-compat --with-file-aio --with-threads --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-mail --with-mail_ssl_module --with-stream --with-stream_realip_module --with-stream_ssl_module --with-stream_ssl_preread_module --with-cc-opt='-g -O2 -fdebug-prefix-map=/data/builder/debuild/nginx-1.18.0/debian/debuild-base/nginx-1.18.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fPIC' --with-ld-opt='-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie'
INFO:__main__:started nginx (pid: 52)
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "GET /ping HTTP/1.1" 200 0 "-" "Go-http-client/1.1"
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "GET /execution-parameters HTTP/1.1" 404 22 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b5a810>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe17b5a810>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b5a810>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b5a810>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e25690>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e25690>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e25690>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e25690>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b38ad0>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe17b38ad0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b38ad0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe17b38ad0>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2cf10>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e2cf10>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2cf10>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2cf10>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
input handler called
INFO:tfs_utils:sagemaker tfs attributes: 
{}
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b790>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e3b790>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b790>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b790>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
input handler called
INFO:tfs_utils:sagemaker tfs attributes: 
{}
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b4d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e3b4d0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b4d0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e3b4d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2ccd0>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e2ccd0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2ccd0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2ccd0>: Failed to establish a new connection: [Errno 111] Connection refused'))
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
INFO:python_service:http://gunicorn_upstream/invocations
INFO:tfs_utils:sagemaker tfs attributes: 
{}
input handler called
json image produced
ERROR:python_service:exception handling request: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e1ba10>: Failed to establish a new connection: [Errno 111] Connection refused'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python3.7/site-packages/gevent/_socketcommon.py", line 607, in connect
    raise _SocketError(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
169.254.255.130 - - [14/Mar/2021:23:56:26 +0000] "POST /invocations HTTP/1.1" 500 283 "-" "Go-http-client/1.1"
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7efe16e1ba10>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e1ba10>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sagemaker/python_service.py", line 292, in _handle_invocation_post
    res.body, res.content_type = self._handlers(data, context)
  File "/sagemaker/python_service.py", line 325, in handler
    response = requests.post(context.rest_uri, data=processed_input)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e1ba10>: Failed to establish a new connection: [Errno 111] Connection refused'))
2021-03-14 23:56:27.554814: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:190] Running initialization op on SavedModel bundle at path: /opt/ml/model/export/Servo/1
2021-03-14 23:56:27.901896: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:277] SavedModel load for tags { serve }; Status: success: OK. Took 3161163 microseconds.
2021-03-14 23:56:27.987438: I tensorflow_serving/servables/tensorflow/saved_model_warmup_util.cc:59] No warmup data file found at /opt/ml/model/export/Servo/1/assets.extra/tf_serving_warmup_requests
2021-03-14 23:56:27.988975: I tensorflow_serving/util/retrier.cc:46] Retrying of Loading servable: {name: Servo version: 1} exhausted max_num_retries: 0
2021-03-14 23:56:27.989006: I tensorflow_serving/core/loader_harness.cc:87] Successfully loaded servable version {name: Servo version: 1}
2021-03-14 23:56:27.991672: I tensorflow_serving/model_servers/server.cc:371] Running gRPC ModelServer at 0.0.0.0:10000 ...
[warn] getaddrinfo: address family for nodename not supported
2021-03-14 23:56:27.992764: I tensorflow_serving/model_servers/server.cc:391] Exporting HTTP/REST API at:localhost:10001 ...
[evhttp_server.cc : 238] NET_LOG: Entering the event loop ...

2021-03-14T23:56:26.474:[sagemaker logs]: MaxConcurrentTransforms=1, MaxPayloadInMB=1, BatchStrategy=MULTI_RECORD
2021-03-14T23:56:26.572:[sagemaker logs]: cvmodel/images/123_1024x768.jpg: Bad HTTP status received from algorithm: 500
2021-03-14T23:56:26.573:[sagemaker logs]: cvmodel/images/123_1024x768.jpg: 
2021-03-14T23:56:26.573:[sagemaker logs]: cvmodel/images/123_1024x768.jpg: Message:
2021-03-14T23:56:26.573:[sagemaker logs]: cvmodel/images/123_1024x768.jpg: {"error": "HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e2cf10>: Failed to establish a new connection: [Errno 111] Connection refused'))"}
2021-03-14T23:56:26.645:[sagemaker logs]: cvmodel/images/124_1024x768.jpg: Bad HTTP status received from algorithm: 500
2021-03-14T23:56:26.645:[sagemaker logs]: cvmodel/images/124_1024x768.jpg: 
2021-03-14T23:56:26.645:[sagemaker logs]: cvmodel/images/124_1024x768.jpg: Message:
2021-03-14T23:56:26.645:[sagemaker logs]: cvmodel/images/124_1024x768.jpg: {"error": "HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /v1/models/Servo:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7efe16e1ba10>: Failed to establish a new connection: [Errno 111] Connection refused'))"}
